### PR TITLE
fix(server): remove dots from badge query param to avoid DNS 1123 spec error

### LINKED ISSUE #15507: Badge not found when app name contains a dot
**Describe the bug**

Application badge is not found, if app name contains dot (`.`).

Should be related to the changes in v2.8.4 https://github.com/argoproj/argo-cd/pull/15468

**To Reproduce**

1. Use argocd v2.8.4
2. Create `Application` which containes dot (`.`) in it's name, e.g.: `test.app`
3. Call `https://ARGOCD_URL/api/badge?name=test.app&revision=true` (answer will be 400 not found)

**Expected behavior**

Application name can contains dots, and badge will work as it did in previous releases.

**Version**

v2.8.4

### PR DESCRIPTION
**Pull request by KouWakai**
… to avoid DNS 1123 specification validate

Fixes #15507 

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [x] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [x] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [x] I have added a brief description of why this PR is necessary and/or what this PR solves.

### CODE REVIEW

**Comment by byrneo** [comment]
Hi @crenshaw-dev @KouWakai - just wondering if this fix is ready to be merged?

**Comment by nweisenauer-sap** [comment]
Hey @KouWakai , looks like DCO check fails because a commit is not signed-off. Can you please check [the instructions here](https://github.com/argoproj/argo-cd/pull/15533/checks?check_run_id=16864931411)?
I would also love to see this one merged :)
Thanks!

**Comment by KouWakai** [comment]
> Hey @KouWakai , looks like DCO check fails because a commit is not signed-off. Can you please check [the instructions here](https://github.com/argoproj/argo-cd/pull/15533/checks?check_run_id=16864931411)? I would also love to see this one merged :) Thanks!

Sorry for kept you waiting and thanks for the instruction!
I would love to include Signed-off-by line!

**Comment by KouWakai** [comment]
@nweisenauer-sap 
DCO is okay! Thanks for the remind!

File: server/badge/badge.go [hunk 1]
```diff
@@ -96,7 +96,7 @@ func (h *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 	//Sample url: http://localhost:8080/api/badge?name=123
 	if name, ok := r.URL.Query()["name"]; ok && enabled && !notFound {
-		if errs := validation.NameIsDNSLabel(strings.ToLower(name[0]), false); len(errs) == 0 {
+		if errs := validation.NameIsDNSLabel(strings.Replace(strings.ToLower(name[0]), ".", "", -1), false); len(errs) == 0 {
```
Comment (gdsoumya) [inline server/badge/badge.go:99]: Can you add a comment here to explain why this is needed.

Comment (KouWakai) [reply server/badge/badge.go:99]: Thanks for the review!!

Sure I'm gonna add a explanation for it

**Review by gdsoumya — APPROVED** [review]
LGTM

**Comment by pepe9012** [comment]
@gdsoumya hi, any chance of merging this PR anytime soon? :)

**Comment by gdsoumya** [comment]
Tagging @jannfis for review as he worked on this before.

**Review by pasha-codefresh — APPROVED** [review]

**Comment by SamuraiPrinciple** [comment]
Is there any chance this could get merged?

**Comment by pasha-codefresh** [comment]
@jannfis could you please also take a look ?

**Comment by agaudreault** [comment]
I created https://github.com/argoproj/argo-cd/pull/17580 to fix the validation with existing methods. It will fix the namespace validation as well.

**Comment by jannfis** [comment]
Sorry for the late reply. IMHO, the better solution would be to fix the validation method used. The idea behind using the DNS label validation would be to validate if the name is a valid Kubernetes resource name, because the name refers to an Application resource. Seems I chose the wrong method initially.

**Pull request closed without merging**